metacat-main/src/main/java/com/netflix/metacat/main/services/impl/ConnectorTableServiceProxy.java

```diff
@@ -0,0 +1,229 @@
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package com.netflix.metacat.main.services.impl;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.netflix.metacat.common.MetacatRequestContext;
+import com.netflix.metacat.common.QualifiedName;
+import com.netflix.metacat.common.dto.TableDto;
+import com.netflix.metacat.common.server.connectors.ConnectorRequestContext;
+import com.netflix.metacat.common.server.connectors.ConnectorTableService;
+import com.netflix.metacat.common.server.converter.ConverterUtil;
+import com.netflix.metacat.common.server.util.MetacatContextManager;
+import com.netflix.metacat.main.manager.ConnectorManager;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.cache.annotation.CacheConfig;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Handles calls to the connector table service.
+ */
+@Slf4j
+@CacheConfig(cacheNames = "metacat")
+public class ConnectorTableServiceProxy {
+    private final ConnectorManager connectorManager;
+    private final ConverterUtil converterUtil;
+
+    /**
+     * Constructor.
+     *
+     * @param connectorManager    connector manager
+     * @param converterUtil       utility to convert to/from Dto to connector resources
+     */
+    public ConnectorTableServiceProxy(
+        final ConnectorManager connectorManager,
+        final ConverterUtil converterUtil
+    ) {
+        this.connectorManager = connectorManager;
+        this.converterUtil = converterUtil;
+    }
+
+    /**
+     * Calls the connector table service create method.
+     * @param name table name
+     * @param tableDto table object
+     */
+    public void create(final QualifiedName name, final TableDto tableDto) {
+        final MetacatRequestContext metacatRequestContext = MetacatContextManager.getContext();
+        final ConnectorTableService service = connectorManager.getTableService(name);
+        final ConnectorRequestContext connectorRequestContext = converterUtil.toConnectorContext(metacatRequestContext);
+        service.create(connectorRequestContext, converterUtil.fromTableDto(tableDto));
+    }
+
+    /**
+     * Calls the connector table service delete method.
+     * @param name table name
+     */
+    @CacheEvict(key = "'table.' + #name")
+    public void delete(final QualifiedName name) {
```

Comment:
How will the deletion failure be handled during @CacheEvict?

Reply:
Cache will not evict if there is a failure.